### PR TITLE
Selective saving for map and location

### PR DESCRIPTION
**Pull request by aronwk-aaron**
Don't save the map and char location info if we are in an instanced Zone

tested going to a prop and making sure everything else saved but I got put back in the origin world.

### CODE REVIEW

File: dGame/dComponents/ControllablePhysicsComponent.cpp [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #include "CDClientManager.h"
 #include "EntityManager.h"
 #include "Character.h"
+#include "dZoneManager.h"
```
Comment (aronwk-aaron) [inline dGame/dComponents/ControllablePhysicsComponent.cpp:14]: My only concern is if this was the best way to be able to check if we are in a zone we don't want t save in from the component.

File: dGame/Character.cpp [hunk 2]
```diff
@@ -301,20 +301,22 @@ void Character::SaveXMLToDatabase() {
...
-			lzidConcat = (lzidConcat << 16) | uint16_t(zoneInfo.GetMapID());
-			character->SetAttribute("lzid", lzidConcat);
-			character->SetAttribute("lnzid", GetLastNonInstanceZoneID());
-
-			//Darwin's backup:
-			character->SetAttribute("lwid", Game::server->GetZoneID());
-
-			// Set the target scene, custom attribute
-			character->SetAttribute("tscene", m_TargetScene.c_str());
+		if ((zoneInfo.GetMapID() != 0) && !(1700 > zoneInfo.GetMapID() && zoneInfo.GetMapID() > 1600)) {
```
Comment (EmosewaMC) [inline dGame/Character.cpp:306]: lup worlds have a should save position flag in the database.  use this instead here.  This comparison is just comparing magic numbers and is not ideal.

**Review by EmosewaMC — CHANGES_REQUESTED** [review]

File: dGame/Character.cpp (hunk 2 above)
Comment (aronwk-aaron) [reply dGame/Character.cpp:306]: oh yeah, there's the whole LUPZoneIDs table

File: dGame/Character.cpp [hunk 3]
```diff
@@ -301,20 +301,22 @@ void Character::SaveXMLToDatabase() {
...
-			character->SetAttribute("lzid", lzidConcat);
-			character->SetAttribute("lnzid", GetLastNonInstanceZoneID());
-
-			//Darwin's backup:
-			character->SetAttribute("lwid", Game::server->GetZoneID());
-
-			// Set the target scene, custom attribute
-			character->SetAttribute("tscene", m_TargetScene.c_str());
+		if ((zoneInfo.GetMapID() != 0)) {
+			if (zoneInfo.GetCloneID() == 0) {
```
Comment (EmosewaMC) [inline dGame/Character.cpp:307]: This can be simplified to if zoneInfo.getMapID != 0 && getCloneID == 0

**Review by EmosewaMC — CHANGES_REQUESTED** [review]
Some simplification of logic gate.  This applies to both sets of changes

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**